### PR TITLE
fix(vm): start on creation for AlwaysOnUnlessStoppedManually

### PR DESCRIPTION
**Pull request by diafour**
## Description

Use annotation to indicate that VM should be started on creation for AlwaysOnUnlessStoppedManualy.

## Why do we need it, and what problem does it solve?

VM should start on creation by design.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/virtualization-artifact/pkg/controller/vm_reconciler.go [hunk 1]
```diff
@@ -735,7 +758,26 @@ func (r *VMReconciler) syncPowerState(ctx context.Context, state *VMReconcilerSt
 		// kubevirt restarts VM via re-creation of KVVMI.
 		err = state.EnsureRunStrategy(ctx, virtv1.RunStrategyAlways)
 	case virtv2.AlwaysOnUnlessStoppedManualy:
-		if state.KVVMI != nil && state.KVVMI.DeletionTimestamp == nil {
+		if state.KVVM.Status.PrintableStatus == virtv1.VirtualMachineStatusStopped {
+			_, hasAnno := state.KVVM.GetAnnotations()["virtualization.deckhouse.io/vm.powerstate.start-on-creation"]
```
Comment (yaroslavborbat) [inline images/virtualization-artifact/pkg/controller/vm_reconciler.go:762]: move annotation to common consts

File: images/virtualization-artifact/pkg/controller/vm_reconciler.go [hunk 2]
```diff
@@ -735,7 +758,26 @@ func (r *VMReconciler) syncPowerState(ctx context.Context, state *VMReconcilerSt
...
+			if hasAnno {
+				// Add start request to start newly created VM.
+				opts.Log.Info("Patch KVVM to delete start-on-creation annotation")
+				patch := `[{"op":"remove", "path": "/metadata/annotations/virtualization.deckhouse.io~1vm.powerstate.start-on-creation"}]`
```
Comment (yaroslavborbat) [inline images/virtualization-artifact/pkg/controller/vm_reconciler.go:766]: we have pkg with [JsonPatch](https://github.com/deckhouse/virtualization/blob/cd942696adfba67ff6da62358922bf27a1cdc6e1/images/virtualization-artifact/pkg/common/patch/patch.go#L14)

**Review by yaroslavborbat — APPROVED** [review]

File: images/virtualization-artifact/pkg/controller/vm_reconciler.go [hunk 3]
```diff
@@ -23,6 +23,7 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 
 	vmutil "github.com/deckhouse/virtualization-controller/pkg/common/vm"
+	common "github.com/deckhouse/virtualization-controller/pkg/controller/common"
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/vm_reconciler.go:26]: redundant import alias

File: images/virtualization-artifact/pkg/controller/vm_reconciler.go [hunk 4]
```diff
@@ -495,6 +511,13 @@ func (r *VMReconciler) createKVVM(ctx context.Context, state *VMReconcilerState,
 		return fmt.Errorf("prepare to create KubeVirt VM '%s': %w", kvvm.GetName(), err)
 	}
 
+	// Workaround for AlwaysOnUnlessStoppedManually
+	if state.VM.Current().Spec.RunPolicy == virtv2.AlwaysOnUnlessStoppedManualy {
+		// Save "newly created" status to annotations to start VM a slightly later.
+		opts.Log.Info("Add annotation to KVVM to start it later after creation")
+		common.AddAnnotation(kvvm, "virtualization.deckhouse.io/vm.powerstate.start-on-creation", "true")
```
Comment (Isteb4k) [inline images/virtualization-artifact/pkg/controller/vm_reconciler.go:518]: It's better to move annotation to const

**Review by Isteb4k — CHANGES_REQUESTED** [review]

**Pull request closed without merging**